model/src/main/resources/db-update/30200-build-push.sql

```diff
@@ -0,0 +1,160 @@
+
+--                    Table "operation_parameters"
+--     Column    |          Type           | Collation | Nullable | Default
+-- --------------+-------------------------+-----------+----------+---------
+--  operation_id | bigint                  |           | not null |
+--  value        | character varying(8192) |           | not null |
+--  key          | character varying(50)   |           | not null |
+
+--                     Table "buildpushreport"
+--    Column    |          Type          | Collation | Nullable | Default
+-- -------------+------------------------+-----------+----------+---------
+-- brewbuildid  | integer                |           | not null |
+-- brewbuildurl | character varying(255) |           |          |
+-- operation_id | bigint                 |           | not null |
+
+
+BEGIN; -- New and updated tables
+
+    ALTER table operation ADD build_id bigint;
+    ALTER TABLE operation
+        ADD CONSTRAINT fk_operation_buildrecord FOREIGN KEY (build_id) REFERENCES buildrecord(id);
+
+    CREATE TABLE buildpushreport
+    (
+        brewbuildid  integer NOT NULL,
+        brewbuildurl character varying(255),
+        operation_id bigint  NOT NULL
+    );
+    ALTER TABLE buildpushreport
+        ADD CONSTRAINT buildpushreport_pkey PRIMARY KEY (operation_id);
+    ALTER TABLE buildpushreport
+        ADD CONSTRAINT fk_buildpushreport_operation FOREIGN KEY (operation_id) REFERENCES operation(id);
+
+COMMIT;
+
+BEGIN; -- Data migration
+
+    INSERT INTO operation
+    SELECT
+        'BuildPush' AS operation_type,
+        bpr.id AS id,
+        to_timestamp(((bpr.id >> 22) + 1577836800000 + 1800000)::double precision/1000) AS endtime,
+        'FINISHED' AS progressstatus,
+        CASE WHEN bpr.status = 'ACCEPTED' THEN 'TIMEOUT'
+             WHEN bpr.status = 'SUCCESS' THEN 'SUCCESSFUL'
+             WHEN bpr.status = 'REJECTED' THEN 'REJECTED'
+             WHEN bpr.status = 'FAILED' THEN 'FAILED'
+             WHEN bpr.status = 'SYSTEM_ERROR' THEN 'SYSTEM_ERROR'
+             WHEN bpr.status = 'CANCELED' THEN 'CANCELLED'
+             ELSE 'SYSTEM_ERROR'
+        END as result,
+        to_timestamp(((bpr.id >> 22) + 1577836800000 + 1)::double precision / 1000) AS starttime,
+        to_timestamp(((bpr.id >> 22) + 1577836800000 + 0)::double precision/1000) AS submittime,
+        CASE WHEN bpr.userinitiator IS NULL THEN (SELECT id FROM usertable WHERE username = 'pnc-admin')
+             ELSE (SELECT id FROM usertable WHERE username = bpr.userinitiator)
+        END as user_id,
+        NULL As productmilestone_id,
+        bpr.buildrecord_id AS build_id
+    FROM buildrecordpushresult bpr
+    WHERE bpr.id > 10000000000000000;
+
+    -- workaround for pre-base32 ids
+    INSERT INTO operation
+    SELECT
+        'BuildPush' AS operation_type,
+        -- this id will not match the time, but at least keeps original order
+        (((bpr.id % 100000000) + 529820) << 22) + floor(random()*4194304) AS id,
```

Comment:
I would maybe add a cast to ensure type consistency:
`(((bpr.id % 100000000) + 529820) << 22) + floor(random()*4194304)::bigint`